strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java

```diff
@@ -68,69 +60,34 @@
     @Inject
     private RepositoryPathResolver repositoryPathResolver;
 
-
-    @BeforeAll
-    public static void cleanUp()
-            throws Exception
-    {
-        cleanUp(getRepositoriesToClean());
-    }
-
-    public static Set<RepositoryDto> getRepositoriesToClean()
-    {
-        Set<RepositoryDto> repositories = new LinkedHashSet<>();
-        repositories.add(createRepositoryMock(STORAGE, REPOSITORY, RawLayoutProvider.ALIAS));
-
-        return repositories;
-    }
-
-    @BeforeEach
-    public void setUp()
-            throws Exception
-    {
-        Configuration configuration = configurationManagementService.getConfiguration();
-
-        if (configuration.getStorage(STORAGE) == null)
-        {
-            createStorage(STORAGE);
-        }
-
-        if (configuration.getStorage(STORAGE).getRepository(REPOSITORY) == null)
-        {
-            createRepository(STORAGE, REPOSITORY);
-        }
-    }
-
-    @AfterEach
-    public void removeRepositories()
-            throws IOException, JAXBException
-    {
-        removeRepositories(getRepositoriesToClean());
-    }
-
     @Test
-    public void testNullArtifactCoordinates()
+    void testNullArtifactCoordinates()
     {
         NullArtifactCoordinates coordinates = new NullArtifactCoordinates("foo/bar/blah.bz2");
 
-        System.out.println("coordinates.toPath(): " + coordinates.toPath());
+        logger.info("coordinates.toPath(): {}", coordinates.toPath());
     }
 
+    @ExtendWith({RepositoryManagementTestExecutionListener.class,
+                 ArtifactManagementTestExecutionListener.class })
     @Test
-    public void testDeployAndResolveArtifact()
+    void testDeployAndResolveArtifact(@TestRepository(layout = RawLayoutProvider.ALIAS,
+                                                      storageId = STORAGE,
+                                                      repositoryId = REPOSITORY)
+                                     Repository repository,
+                                     @TestArtifact(storageId = STORAGE,
+                                                   repositoryId = REPOSITORY,
+                                                   resource = "foo/bar.zip",
+                                                   generator = NullArtifactGenerator.class)
+                                     Path artifactPath)
             throws Exception
     {
         String path = "foo/bar.zip";
 
-        RepositoryPath repositoryPath = repositoryPathResolver.resolve(STORAGE, REPOSITORY, path);
-
-        // Deploy the artifact
-        artifactManagementService.validateAndStore(repositoryPath, createZipFile());
+        RepositoryPath repositoryPath = repositoryPathResolver.resolve(repository, path);
```

Comment:
we shouldn't resolve path here, just use path that you have injected (note that it shouldn't be deployed).

Reply:
If I do this:
```
@TestArtifact(resource = "foo/bar.zip",
              generator = NullArtifactGenerator.class)
Path artifactPath
```

I need a `RepositoryPath` to use it as 1st parameter in `validateAndStore` method.
Then I can't even use `artifactPath.toString()` (it returns an absolute path), which does not give me `foo/bar.zip`, which is what I need.

The only working code that I know of is:
```
String path = "foo/bar.zip";
RepositoryPath repositoryPath = repositoryPathResolver.resolve(repository, path);
```

Any ideas?

Reply:
right, sorry for confusion